packages/vite/src/node/optimizer/optimizer.ts

```diff
@@ -754,33 +720,43 @@ async function createDepsOptimizer(
       debouncedProcessing(0)
     }
   }
-}
 
-async function createDevSsrDepsOptimizer(
-  config: ResolvedConfig,
-): Promise<void> {
-  const metadata = await optimizeServerSsrDeps(config)
+  return depsOptimizer
+}
 
+export function createExplicitDepsOptimizer(
+  environment: DevEnvironment,
+): DepsOptimizer {
   const depsOptimizer = {
-    metadata,
-    isOptimizedDepFile: createIsOptimizedDepFile(config),
-    isOptimizedDepUrl: createIsOptimizedDepUrl(config),
+    metadata: initDepsOptimizerMetadata(environment),
+    isOptimizedDepFile: createIsOptimizedDepFile(environment),
+    isOptimizedDepUrl: createIsOptimizedDepUrl(environment),
     getOptimizedDepId: (depInfo: OptimizedDepInfo) =>
       `${depInfo.file}?v=${depInfo.browserHash}`,
 
     registerMissingImport: () => {
       throw new Error(
-        'Vite Internal Error: registerMissingImport is not supported in dev SSR',
+        `Vite Internal Error: registerMissingImport is not supported in dev ${environment.name}`,
       )
     },
+    init,
     // noop, there is no scanning during dev SSR
     // the optimizer blocks the server start
     run: () => {},
 
     close: async () => {},
-    options: config.ssr.optimizeDeps,
+    options: environment.config.dev.optimizeDeps,
```

Comment:
`environment.options` instead of `environment.config` since `config` is the entire vite config?